webapp/channels/src/components/cloud_preview_modal/preview_modal_controller.test.tsx

```diff
@@ -0,0 +1,248 @@
+        children: React.ReactNode;
+        show: boolean;
+        onHide?: () => void;
+        showCloseButton?: boolean;
+        footerContent?: React.ReactNode;
+    }) => (
+        show ? (
+            <div data-testid='modal'>
+                {showCloseButton && (
+                    <button
+                        aria-label='Close'
+                        onClick={onHide}
+                        data-testid='close-button'
+                    >
+                        <span data-testid='close-icon'>{'24'}</span>
+                    </button>
+                )}
+                <div>{children}</div>
+                {footerContent && <div data-testid='modal-footer'>{footerContent}</div>}
+            </div>
+        ) : null
+    ),
+}));
+
+// Mock the icons
+jest.mock('@mattermost/compass-icons/components', () => ({
+    ArrowLeftIcon: ({size}: {size: number}) => <span data-testid='arrow-left-icon'>{size}</span>,
+    ArrowRightIcon: ({size}: {size: number}) => <span data-testid='arrow-right-icon'>{size}</span>,
+    CloseIcon: ({size}: {size: number}) => <span data-testid='close-icon'>{size}</span>,
+}));
+
+describe('PreviewModalController', () => {
+    const mockOnClose = jest.fn();
+
+    const contentData: PreviewModalContentData[] = [
+        {
+            title: {
+                id: 'test.slide1.title',
+                defaultMessage: 'First Slide',
+            },
+            subtitle: {
+                id: 'test.slide1.subtitle',
+                defaultMessage: 'First subtitle',
+            },
+            skuLabel: {
+                id: 'test.slide1.sku_label',
+                defaultMessage: 'First sku label',
+            },
+            videoUrl: 'https://www.youtube.com/watch?v=E3EGLxgNxNA',
+            useCase: 'missionops',
+        },
+        {
+            title: {
+                id: 'test.slide2.title',
+                defaultMessage: 'Second Slide',
+            },
+            subtitle: {
+                id: 'test.slide2.subtitle',
+                defaultMessage: 'Second subtitle',
+            },
+            skuLabel: {
+                id: 'test.slide2.sku_label',
+                defaultMessage: 'Second sku label',
+            },
+            videoUrl: 'https://www.youtube.com/watch?v=E3EGLxgNxNA',
+            useCase: 'missionops',
+        },
+        {
+            title: {
+                id: 'test.slide3.title',
+                defaultMessage: 'Third Slide',
+            },
+            subtitle: {
+                id: 'test.slide3.subtitle',
+                defaultMessage: 'Third subtitle',
+            },
+            skuLabel: {
+                id: 'test.slide3.sku_label',
+                defaultMessage: 'Third sku label',
+            },
+            videoUrl: 'https://www.youtube.com/watch?v=E3EGLxgNxNA',
+            useCase: 'missionops',
+        },
+    ];
+
+    beforeEach(() => {
+        mockOnClose.mockClear();
```

Comment:
This isn't needed because we have Jest configured to automatically clear the mocks